### ISSUE TITLE
Rule for chained class names

### ISSUE THREAD

**Issue by evisong** [stylelint/stylelint#2491]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I'm not sure if this is a bug. 

According to http://stackoverflow.com/a/2554853/245345, sometimes selectors like `.class1.class2` are used for selecting elements that have both classes specified (`class="class1 class2"`). In latest version of Stylelint, this `.class1.class2` is treated as a single class name, and is applicable to the rules like `selector-class-pattern`.

Is this better to be considered a separate use case?

> Which rule, if any, is this issue related to?

`selector-class-pattern`

> What CSS is needed to reproduce this issue?

```css
.class1.class2 {
    /* style here */
}
```

> What stylelint configuration is needed to reproduce this issue?

I use the rule below to recommend dash-only style class names:

```json
{
  "rules": {
    "selector-class-pattern": ["^[a-z]+([a-z0-9]?|[a-z0-9\\-]*[a-z0-9])$", {
        "message": "Please name CSS classes like this: 'my-1st-class-name'",
        "severity": "warning"
      }]
  }
}
```

> Which version of stylelint are you using?

e.g. `7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No, this should be part of CSS standard: https://www.w3.org/TR/CSS2/selector.html#class-html

> To match a subset of "class" values, each value must be preceded by a ".".
> 
> For example, the following rule matches any P element whose "class" attribute has been assigned a list of space-separated values that includes "pastoral" and "marine":
> 
> 
> p.marine.pastoral { color: green }
> This rule matches when class="pastoral blue aqua marine" but does not match for class="pastoral blue".

> What did you expect to happen?

Each of `class1` and `class2` is verified by `selector-class-pattern` rule separately.

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
39:1   ⚠  Please name CSS classes like this: 'my-1st-class-name'  selector-class-pattern
```

Thanks.

**Comment by evisong** [comment]
Sorry my bad, my test case contains uppercase characters. It's already supported.
